pkg/utils/gardener/shoot.go

```diff
@@ -736,6 +736,12 @@ func ComputeRequiredExtensionsForShoot(shoot *gardencorev1beta1.Shoot, seed *gar
 	return requiredExtensions
 }
 
+func extensionEnabledForMode(extensionID string, enableMode gardencorev1beta1.AutoEnableMode, resource gardencorev1beta1.ControllerResource, disabledExtensions sets.Set[string]) bool {
```

Comment:
Nit: This function is also used for the `Seed`, perhaps move it out of `shoot.go`?